src/analysis/souffle/list_macros.dl

```diff
@@ -0,0 +1,7 @@
+#ifndef LIST_MACROS_H__
+#define LIST_MACROS_H__
+
+#define LIST_TYPE(EltType) EltType##List
+#define DEFINE_LIST_TYPE(EltType) .type LIST_TYPE(EltType) [ item: EltType, next: LIST_TYPE(EltType) ]
```

Comment:
This is nice. I didn't know you could also do type definitions this way in Souffle.

Reply:
I'm not going to claim this is exactly right; I didn't compile it. But since Souffle runs the C Preprocessor, it should be the case that C Preprocessor-type tricks should work similarly to how they do in a C/C++ file, similar to how they work in .S (preprocessing-needed) assembly files.